### PR TITLE
chore: Check paradox for pr validation.

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation:
Check the paradox too.

refs: https://github.com/apache/incubator-pekko/issues/959

### CODE REVIEW

File: .github/workflows/generate-doc-check.yml [hunk 1]
```diff
@@ -50,3 +50,6 @@ jobs:
 
       - name: Compile docs for all Scala versions
         run: sbt +compile:doc
+
+      - name: Compile paradox docs
+        run: sbt paradox
```
Comment (pjfanning) [inline .github/workflows/generate-doc-check.yml:55]: do we need `sbt +compile:doc` if we do `sbt paradox`?

Comment (He-Pin) [reply .github/workflows/generate-doc-check.yml:55]: Seems like it would be better to chain the commands.

Comment (pjfanning) [reply .github/workflows/generate-doc-check.yml]: it's not a necessarily good idea to run separate commands. You will often find that the work is done from scratch, on each call.

File: .github/workflows/generate-doc-check.yml [hunk 2]
```diff
@@ -49,4 +49,4 @@ jobs:
           sudo apt-get install graphviz
 
       - name: Compile docs for all Scala versions
-        run: sbt +compile:doc
+        run: sbt ";+compile:doc; paradox"
```
Comment (mdedetrich) [inline .github/workflows/generate-doc-check.yml:52]: Does this need to be `+paradox`

Comment (pjfanning) [reply .github/workflows/generate-doc-check.yml:52]: I think we only need to build this once - with java 11. Paradox doesn't work with Java 17 (you need to add JPMS flags to the java command at least).

**Review by mdedetrich — APPROVED** [review]
lgtm, lets double check this before merging by creating a deliberate compilation error in one of the paradox code sources to see if it fails

**Comment by He-Pin** [comment]
I tested it locally , it works
image

**Comment by mdedetrich** [comment]
Perfect, feel free to merge it once you verify with the `test` commit and remove it.

Also cherry pick it to the `1.0.x` branch once merged

**Comment by He-Pin** [comment]
@mdedetrich No sure why ,but even with dedicated tests, it can not detecting anything.

**Comment by mdedetrich** [comment]
I'll try and look into it today

**Comment by He-Pin** [comment]
The `docs/Test/compile` can't catch the compiling error either, wired.

**Comment by Roiocam** [comment]
> @mdedetrich No sure why ,but even with dedicated tests, it can not detecting anything.

Pekko already has build script for those` java-jdk9-only` directory, https://github.com/apache/incubator-pekko/blob/main/project/Jdk9.scala

But you have to run test/compile on JDK11 or higher.

**Comment by Roiocam** [comment]
This could be closed because the problem has been found, see #959 and related issues for more details.

**Comment by He-Pin** [comment]
@Roiocam Thanks for update.

**Pull request closed without merging**